### ISSUE TITLE
Agda agda-cubical to the Agda test suite

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#4445]
To ensure we are not accidentially introduce bugs in `--cubical` we ought to have the cubical library as standard test, just like the standard library.
https://github.com/agda/cubical

**Comment by jespercockx** [comment]
Could we have a command to automatically fetch the right version of the cubical library in the makefile, similar to `make up-to-date-stdlib`?

**Comment by UlfNorell** [comment]
I added cubical to the `std-lib` targets, so `make up-to-date-stdlib` should update `cubical` as well.

**Comment by asr** [comment]
I suggest to add independents targets for cubical, something like

 ```bash
.PHONY : cubical ## Update the cubical library.
cubical :
	git submodule update --init cubical

.PHONY : up-to-date-cubical ##
up-to-date-cubical : cubical

.PHONY : fast-forward-cubical ##
fast-forward-cubical :
	git submodule update --init --remote cubical
```

**Comment by UlfNorell** [comment]
Why? What's a use case where you want to update one but not the other?

**Comment by asr** [comment]
> Why?

1. Cubical doesn't depend of the standard library.

2. We don't have a defined workflow for updating the standard library in the Agda repo which cause some problems. These problems are independents of cubical.

3. The standard library depends of GHC, some Haskell libraries and Cabal but cubical doesn't.

> What's a use case where you want to update one but not the other?

Testing new versions of GHC and Cabal.

**Comment by UlfNorell** [comment]
The targets only make sure that the submodules are checked out at the right commit. I don't see why any of these use cases require cubical to be out of sync.

**Comment by asr** [comment]
Fact 1: We use the standard library in various tests (succeed, benchmark, etc). 

Hypothesis 1: We shall use also the cubical library in various tests.

Fact 2: The current commit of the standard library in the Agda repository is behind the current commit in the standard library repository.

Hypothesis 2: The current commit of the cubical library in the Agda repository will be behind the current commit in the cubical library repository.

Some possible users of the `Makefile` are:

1. Agda developer who needs to test your changes to Agda on the current versions of the standard library and the cubical library.

2. Standard library developer who needs to test your changes to the standard library on Agda upstream.

3. Cubical library developer who wants to test your changes to the cubical library on Agda upstream.

4. Agda releaser who needs to test the current version of the standard library as preparation for the release.

5. Agda releaser who needs to test the current version of the cubical library as preparation for the release.

The current target `fast-forward-std-lib` is only designed for user 1. For users 2-5 this target could cause unnecessary complications.

**Comment by asr** [comment]
I was planning to update the submodule commit of the standard library. After running `make fast-forward-std-lib`, I got

```
$ git status 
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   cubical (new commits)
	modified:   std-lib (new commits)
```

Note that the cubical library has new commits but I wasn't planning to run the cubical library tests. I'll add independent targets for the cubical library.

**Comment by asr** [comment]
Travis CI is [failing](https://travis-ci.org/agda/agda/builds/658716950) when testing the cubical library on the release branches. I'll write a fix.

**Comment by UlfNorell** [comment]
Do you not need to update `.travis/cabal_install` to also fetch cubical now?

**Comment by asr** [comment]
> Do you not need to update `.travis/cabal_install` to also fetch cubical now?

Testing right now. I don't understand why this issue only shows up in the "complete tests".

**Comment by asr** [comment]
> I don't understand why this issue only shows up in the "complete tests".

Because the "complete-job" doesn't use

```
git:
        submodules: true
```

**Comment by asr** [comment]
Oops, fixed by 4328ecefa0c697fdce258cbaebd9c9514544bb04.
